### ISSUE TITLE
MISSING_TO_STRING_OVERRIDE with invalid field

### ISSUE THREAD

**Issue by jsotuyod** [Monits/findbugs-plugin#2]
The folllowing class is reported as MISSING_TO_STRING_OVERRIDE

```
public class FSTSerializationStrategy implements SerializationStrategy {
	private final FSTConfiguration config;
        ...
```

This shouldn't happen as:
 1. `FSTConfiguration` doesn't override `toString`
 2. `FSTConfiguration` is a library class, and therefore, we can't add the missing `toString` to it.

The current detector should ignore fields that are not part of the application being analyzed, and do not override `toString`.

Possibly, the first condition can be tested using [`AnalysisContext.isApplicationClass()`](https://code.google.com/p/findbugs/source/browse/findbugs/src/java/edu/umd/cs/findbugs/ba/AnalysisContext.java#433), though I'm not sure.
